zipline/_protocol.pyx

```diff
@@ -482,6 +487,9 @@ cdef class BarData:
         cdef object session_label
         cdef object dt_to_use_for_exchange_check,
 
+        if self._rl_controller.is_restricted(asset, adjusted_dt):
```

Comment:
We might want to cache the method lookup here at the constructor.  That will save us an attribute access per invocation here.